smi-test/main.go

```diff
@@ -0,0 +1,294 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	harness "github.com/kudobuilder/kuttl/pkg/apis/testharness/v1beta1"
+	"github.com/kudobuilder/kuttl/pkg/test"
+	testutils "github.com/kudobuilder/kuttl/pkg/test/utils"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/discovery"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type URLstruct struct {
+	URL     string
+	Method  string
+	Headers map[string]string
+}
+type MetricResponse struct {
```

Comment:
Does it make sense to instrument using https://github.com/prometheus/client_golang?  // @kanishkarj